lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -1997,6 +2027,112 @@ SPIRVValue *LLVMToSPIRV::transIntrinsicInst(IntrinsicInst *II,
+    std::vector<SPIRVValue *> Args{transValue(II->getArgOperand(0), BB),
+                                   transValue(II->getArgOperand(1), BB),
+                                   transValue(II->getArgOperand(2), BB)};
+    auto UnDecBM = BM->addExtInst(transType(II->getType()),
+                                  BM->getExtInstSetId(SPIRVEIS_OpenCL),
+                                  OpenCLLIB::Fma, Args, BB);
+    return applyRoundingModeConstrain(II->getOperand(3), UnDecBM);
+  }
+  case Intrinsic::experimental_constrained_fptoui: {
+    return BM->addUnaryInst(OpConvertFToU, transType(II->getType()),
+                            transValue(II->getArgOperand(0), BB), BB);
+  }
+  case Intrinsic::experimental_constrained_fptosi: {
+    return BM->addUnaryInst(OpConvertFToS, transType(II->getType()),
+                            transValue(II->getArgOperand(0), BB), BB);
+  }
+  case Intrinsic::experimental_constrained_uitofp: {
+    auto UnDecBM = BM->addUnaryInst(OpConvertUToF, transType(II->getType()),
+                                    transValue(II->getArgOperand(0), BB), BB);
+    return applyRoundingModeConstrain(II->getOperand(1), UnDecBM);
+  }
+
+  case Intrinsic::experimental_constrained_sitofp: {
+    auto UnDecBM = BM->addUnaryInst(OpConvertSToF, transType(II->getType()),
+                                    transValue(II->getArgOperand(0), BB), BB);
+    return applyRoundingModeConstrain(II->getOperand(1), UnDecBM);
+  }
+
+  case Intrinsic::experimental_constrained_fpext: {
+    return BM->addUnaryInst(OpFConvert, transType(II->getType()),
+                            transValue(II->getArgOperand(0), BB), BB);
+  }
+  case Intrinsic::experimental_constrained_fptrunc: {
+    auto UnDecBM = BM->addUnaryInst(OpFConvert, transType(II->getType()),
+                                    transValue(II->getArgOperand(0), BB), BB);
+    return applyRoundingModeConstrain(II->getOperand(1), UnDecBM);
+  }
+
+  case Intrinsic::experimental_constrained_fcmp:
+  case Intrinsic::experimental_constrained_fcmps: {
+    auto MetaMod = cast<MetadataAsValue>(II->getOperand(2))->getMetadata();
+
+    Op CmpTypeOp = StringSwitch<Op>(cast<MDString>(MetaMod)->getString())
+                       .Case("oeq", OpFOrdEqual)
+                       .Case("ogt", OpFOrdGreaterThan)
+                       .Case("oge", OpFOrdGreaterThanEqual)
+                       .Case("olt", OpFOrdLessThan)
+                       .Case("ole", OpFOrdLessThanEqual)
+                       .Case("one", OpFOrdNotEqual)
+                       .Case("ord", OpOrdered)
+                       .Case("ueq", OpFUnordEqual)
+                       .Case("ugt", OpUGreaterThan)
+                       .Case("uge", OpUGreaterThanEqual)
+                       .Case("ult", OpULessThan)
+                       .Case("ule", OpULessThanEqual)
+                       .Case("une", OpFUnordNotEqual)
+                       .Case("uno", OpUnordered)
+                       .Default(OpFAdd);
```

Comment:
```suggestion
                       .Default(OpNop);
```

I think that `nop` serves better as some default value which should be ignored